go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/elastic/elastic-agent
 
-go 1.21.0
-
-toolchain go1.21.10
+go 1.21.9
```

Comment:
Is this intentional ? Do we need to go back to 1.21.9 from 1.21.10 toolchain ?

Reply:
Changing the Go version was not intentional, I'll make it match our `.go-version` file.

Removing the `toolchain go1.21.10` is intentional, it breaks our mage automation that runs in Docker if the local Go version is not exactly the toolchain one.

When we run things in Docker we mount some of the go cache as read-only, but if `toolchain` is defined and the Go version running is different, Go will try to download the toolchain for that version and write to this read-only folder, which fails the build.

I believe the `toolchain` directive was not introduced intentionally.